src/server/controllers/dockerController.ts

```diff
@@ -88,30 +88,31 @@ dockerController.createDockerCompose = (req: Request, res: Response, next: NextF
   let directory: string;
   let containerName: string;
   const composeFilePath = `./myProjects/${projectFolder}/docker-compose.yaml`
-  // making docker compose file
-  // indentation is important in yaml files
-  // checking if compose file already exists. If it does not, it will make one
-  if (!fs.existsSync(composeFilePath)) {
-    // spacing matters so it looks weird on purpose
-    fs.writeFileSync(composeFilePath, `version: "3"\nservices:\n`,
-      (error: Error) => {
-        if (error) return next({
-          log: 'ERROR IN CREATING COMPOSE FILE ',
-          msg: { err: `ERROR: ${error}` }
-        })
-      });
-  }
+  /* writeFile will create a new docker compose file each time the controller is run 
+  so user can have leave-one-out functionality. Indentation is important in yaml files so it looks weird on purpose */
+  fs.writeFileSync(composeFilePath, `version: "3"\nservices:\n`,
+    (error: Error) => {
+      if (error) return next({
+        log: 'ERROR IN CREATING COMPOSE FILE ',
+        msg: { err: `ERROR: ${error}` }
+      })
+    })
+
   // Taking the 'checked' repositories and storing each name into an array
   const { repos } = res.locals;
   const repoArray = [];
   for (const repo of repos) {
     repoArray.push(repo.repoName);
   }
+  // adding service information to docker compose file
   for (let i = 0; i < buildPathArray.length; i++) {
-    // Checking if the array of names includes the repositories stored locally
-    if (!repoArray.includes(containerNameArray[i])) {
-      directory = buildPathArray[i];
-      containerName = containerNameArray[i];
+    directory = buildPathArray[i];
+    containerName = containerNameArray[i];
+    // only gets repos stored in the active Project that have dockerfiles (using buildPath to grab repo folder)
+    const repoFolder = directory.slice(14 + projectFolder.length, directory.length - containerName.length - 1);
```

Comment:
I think this logic can be done cleaner with RegEx. Since you know the level in the file path the repoFolder will be, you can match that pattern the find the fold name. This will also present possible bugs from future developers changing 'myProjects' to something else.

Reply:
It looks like the way you're slicing to get the repoFolder name means that Dockerfiles at the root level of the repository are missed when we create the docker-compose.yaml. This is an issue that will need to be fixed.

![docker-compose1](https://user-images.githubusercontent.com/7851187/87258999-f1252480-c45c-11ea-84d1-a6a63a6292ee.png)
![docker-compose2](https://user-images.githubusercontent.com/7851187/87259010-00a46d80-c45d-11ea-9809-ab3a940773ad.png)
![docker-compose-3](https://user-images.githubusercontent.com/7851187/87259012-05692180-c45d-11ea-8987-0ce0c5fd296e.png)